### PR TITLE
A bunch o' fixes regarding command tree selection.

### PR DESCRIPTION
**Pull request by pmuetschard**
Includes a fix for #938 and #966

### CODE REVIEW

File: gapis/resolve/command_tree.go [hunk 1]
```diff
@@ -140,12 +141,40 @@ func CommandTreeNodeForCommand(ctx context.Context, p *path.CommandTreeNodeForCo
...
+	}
+
 	return &path.CommandTreeNode{
 		Tree:    p.Tree,
-		Indices: cmdTree.indices(api.CmdID(cmdIdx)),
+		Indices: cmdTree.findNodeForCommand(snc, c.Commands, api.CmdID(cmdIdx)),
 	}, nil
 }
 
+func (t *commandTree) findNodeForCommand(snc *sync.Data, cmds []api.Cmd, cmd api.CmdID) []uint64 {
```
Comment (ben-clayton) [inline gapis/resolve/command_tree.go]: I'm really not sure about this change. Please can you move this CL into a separate PR?

File: gapis/resolve/events.go [hunk 2]
```diff
@@ -43,13 +43,24 @@ func Events(ctx context.Context, p *path.Events) (*service.Events, error) {
 	events := []*service.Event{}
 
 	s := c.NewState()
+	lastCmd := api.CmdID(0)
+	delayedFirstInFrame := false
```
Comment (ben-clayton) [inline gapis/resolve/events.go]: This took me a while to understand.

How about:

```go
var pending []*serviceEvent
api.ForeachCmd(ctx, c.Commands, func(ctx context.Context, id api.CmdID, cmd api.Cmd) error {
	cmd.Mutate(ctx, id, s, nil)

	// TODO: Add event generation to the API files.
	if !filter(id, cmd, s) {
		return nil
	}

	events, pending = append(events, pending...), nil
```
...
```go
	if p.FirstInFrame && f.IsEndOfFrame() {
		pending = append(pending, &service.Event{true
			Kind:    service.EventKind_FirstInFrame,		
			Command: p.Capture.Command(uint64(id) + 1),		
		})
	}
```

File: gapis/resolve/command_tree.go (hunk 1 above)
Comment (pmuetschard) [reply gapis/resolve/command_tree.go]: Done.

File: gapis/resolve/events.go (hunk 2 above)
Comment (pmuetschard) [reply gapis/resolve/events.go]: Done.
